pulsar-io/jdbc/core/src/main/java/org/apache/pulsar/io/jdbc/JdbcAbstractSink.java

```diff
@@ -214,11 +214,20 @@ private void flush() {
                             throw new IllegalArgumentException(msg);
                     }
                 }
-                connection.commit();
+                if (!connection.getAutoCommit()) {
```

Comment:
in some JDBC drivers this "connection.getAutoCommit()" executes a RPC (request/response cycle on the wire).
I suggest you to use the configuration entry "useTransactions" to decide to call commit/rollback